mkosi/__init__.py

```diff
@@ -251,13 +260,46 @@ def run(
         MkosiPrinter.info("+ " + " ".join(shlex.quote(x) for x in cmdline))
 
     if not stdout and not stderr:
-        # Unless explicit redirection is done, print all subprocess output on stderr since we do so as well
-        # for mkosi's own output.
+        # Unless explicit redirection is done, print all subprocess
+        # output on stderr, since we do so as well for mkosi's own
+        # output.
+        stdout = sys.stderr
+
+    if stdin is None:
+        stdin = subprocess.DEVNULL
+
+    cm = do_delay_interrupt if delay_interrupt else do_noop
+    try:
+        with cm():
+            return subprocess.run(cmdline, check=check, stdin=stdin, stdout=stdout, stderr=stderr, **kwargs)
+    except FileNotFoundError:
+        die(f"{cmdline[0]} not found in PATH.")
+
+
+def popen(
```

Comment:
I did it because there's a `run()` function, which invokes `subprocess.run()`. So a new function which invokes `subprocess.Popen()` is called `popen()`. But I don't think the `Popen()` name is very good, so I'm happy to rename this. Maybe `spawn` ?

Reply:
I think I like spawn.

Reply:
I rewrote the two functions a bit: now there's spawn() which does all the work, and run() which is a thin wrapper. This way it's not necessary to duplicate the logging and argument mangling logic.

Reply:
I rewrote the wrapper again. I had trouble getting all the stdin/stdout/input/output/stderr/text cases right.